# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1456521218 28800
#      Fri Feb 26 13:13:38 2016 -0800
# Node ID 8cfd1b47d49fbf10a4056d3006dbb5c590c73471
# Parent  1bb9a34011d1785dfdc818b00e558fc61824544c
Eliminate more extra newlines to stdout when not interactive (bug #47258)

* toplev.cc (main_loop): Only print newlines on interrupt or EOF when
running interactively.

diff --git a/libinterp/corefcn/toplev.cc b/libinterp/corefcn/toplev.cc
--- a/libinterp/corefcn/toplev.cc
+++ b/libinterp/corefcn/toplev.cc
@@ -735,17 +735,18 @@ main_loop (void)
       catch (const octave_interrupt_exception&)
         {
           recover_from_exception ();
 
           if (quitting_gracefully)
             return exit_status;
 
           // Required newline when the user does Ctrl+C at the prompt.
-          octave_stdout << "\n";
+          if (interactive)
+            octave_stdout << "\n";
         }
       catch (const index_exception& e)
         {
           recover_from_exception ();
 
           std::cerr << "error: unhandled index exception: "
                     << e.message () << " -- trying to return to prompt"
                     << std::endl;
@@ -779,17 +780,18 @@ main_loop (void)
         {
           if (setjump (naninf_jump) != 0)
             debug_or_throw_exception (true);  // true = stack trace
         }
 #endif
     }
   while (retval == 0);
 
-  octave_stdout << "\n";
+  if (interactive)
+    octave_stdout << "\n";
 
   if (retval == EOF)
     retval = 0;
 
   return retval;
 }
 
 // Fix up things before exiting.
